### PR TITLE
disable video form inputs when editing nft

### LINKED ISSUE #2297: Block Edits for Video for Minted NFT
**AC:**
- [ ] Disable all fields video edit fields as per design
- [ ] Separate VideoForm component into smaller component for input section (split to be figured out)

Block edits in video workspace when NFT is minted for the video

[Originating from this design ticket>](https://github.com/Joystream/atlas/issues/2168)

Figma: https://www.figma.com/file/xm6vBdSnQDroc57rGfGtPM/Video-workspace?node-id=2174%3A75868

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #2297 

**Design**
https://www.figma.com/file/xm6vBdSnQDroc57rGfGtPM/Video-workspace?node-id=1332%3A97483

### CODE REVIEW

File: packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.styles.ts [hunk 1]
```diff
@@ -105,3 +108,16 @@ export const SwitchNftWrapper = styled.div`
...
+export const VideoLink = styled(Link)`
+  font: ${cVar('typographyDesktopT200')};
+  letter-spacing: ${cVar('typographyDesktopT200LetterSpacing')};
+  text-transform: ${cVar('typographyDesktopT200TextTransform')};
+  color: ${cVar('colorCoreBlue300')};
+  text-decoration: none;
+`
+
+export const DescriptionTextArea = styled(TextArea)`
+  opacity: 0.5;
```
Comment (kdembler) [inline packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.styles.ts]: I don't think that's desired?

**Comment by kdembler** [comment]
Also this scenario doesn't work:
1. I open edit for a video that doesn't have an NFT yet
2. I toggle "Mint an NFT"
3. The form becomes uneditable immediately
This should only happen if the NFT has already been minted in the past

File: packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.styles.ts [hunk 2]
```diff
@@ -105,3 +108,15 @@ export const SwitchNftWrapper = styled.div`
...
+export const VideoLink = styled(Link)`
+  font: ${cVar('typographyDesktopT200')};
+  letter-spacing: ${cVar('typographyDesktopT200LetterSpacing')};
+  text-transform: ${cVar('typographyDesktopT200TextTransform')};
+  color: ${cVar('colorCoreBlue300')};
+  text-decoration: none;
+`
+
+export const DescriptionTextArea = styled(TextArea)`
+  margin-top: ${({ disabled }) => disabled && '40px'};
```
Comment (drillprop) [inline packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.styles.ts]: let's use `sizes` here. Also if we leave that, we will have `margin-top: false`, which is not great.

File: packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx [hunk 3]
```diff
@@ -411,10 +446,17 @@ export const VideoForm: React.FC<VideoFormProps> = React.memo(({ onSubmit, setFo
...
                 icon={<StyledSvgWarning width={24} height={24} />}
                 description={
-                  <>
-                    <Text variant="t200">After issuing this as an NFT </Text>
-                    <YellowText variant="t200">editing options of this video will be disabled</YellowText>
-                  </>
+                  !videoFieldsLocked ? (
+                    <>
+                      <Text variant="t200">After issuing this as an NFT </Text>
+                      <YellowText variant="t200">editing options of this video will be disabled</YellowText>
```
Comment (drillprop) [inline packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx]: I think this is my code, but I'm pretty sure that we can put YellowText inside Text i.e. `<Text>< YellowText></Text>`. Could you do that?

File: packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx [hunk 4]
```diff
@@ -411,10 +446,17 @@ export const VideoForm: React.FC<VideoFormProps> = React.memo(({ onSubmit, setFo
...
+                    </>
+                  ) : (
+                    <>
+                      <Text variant="t200">Many fields are disabled after minting an NFT for this video - </Text>
+                      <VideoLink to={absoluteRoutes.viewer.video(editedVideoInfo.id)}>go to it's video page.</VideoLink>
```
Comment (drillprop) [inline packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx]: same as above

**Review by drillprop — CHANGES_REQUESTED** [review]
I still can edit the thumbnail. You need to disable also MultifileSelect. 

Besides that, looks good, found only some minor issues.

**Review by drillprop — APPROVED** [review]

**Pull request merged**